### ISSUE TITLE
`tf.dynamic_stitch` crash with segmentation fault

### ISSUE THREAD

**Issue by creakseek** [tensorflow/tensorflow#58354]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.10

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### CUDA/cuDNN version

n/a

### Current Behaviour?

```shell
`tf.dynamic_stitch` crash with segmentation fault
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
print(tf.__version__)

indices_data = np.array([[4, 3, 1, 7], [1, 5, 1, 0]])
updates_data = np.array([9, 10, 11, 12])
indices = tf.Variable(indices_data, dtype=tf.int32)
updates = tf.Variable(updates_data, dtype=tf.int32)
stitch = tf.dynamic_stitch(indices, updates)
```

### Relevant log output

```shell
2.10.0
segmentation fault
```
</details>

**Comment by bhack** [comment]
Yes it is confirmed in TF 2.10 but it is already solved/protected in TF 2.11.0rc1 so I don't think we are going to cherry-pick this for  2.10 patch release.

```
import os
import tensorflow as tf
import numpy as np
print(tf.__version__)

indices_data = np.array([[4, 3, 1, 7], [1, 5, 1, 0]])
updates_data = np.array([9, 10, 11, 12])
indices = tf.Variable(indices_data, dtype=tf.int32)
updates = tf.Variable(updates_data, dtype=tf.int32)
stitch = tf.dynamic_stitch(indices, updates)
```

```python
InvalidArgumentError: NodeDef expected inputs 'int32, int32, int32, int32' do not match 6 inputs specified; Op<name=DynamicStitch; signature=indices:N*int32, data:N*T -> merged:T; attr=N:int,min=1; attr=T:type>; NodeDef: {{node DynamicStitch}} [Op:DynamicStitch]
```

Of course you can solve this also in TF 2.10 if you respect the documentation constrain for the 2nd arg about the shape:

> data | A list with the same length as indices of Tensor objects with the same type.

**Comment by mihaimaruseac** [comment]
@tilakrayal that's exactly what has been said in the message above yours.

**Comment by bhack** [comment]
Do you mean that it will be backported?

**Comment by mihaimaruseac** [comment]
Unlikely to backport if it doesn't turn out to be a security issue

**Comment by bhack** [comment]
> Unlikely to backport if it doesn't turn out to be a security issue

So as it is solved in 2.11rc I think it could go stale/closed.
